torch/_logging/_internal.py

```diff
@@ -738,9 +741,40 @@ def _default_formatter():
 DEFAULT_FORMATTER = _default_formatter()
 
 
+class TorchLogsFilter(logging.Filter):
+    def __init__(self, ranks):
+        self.ranks = set(ranks)
+        super().__init__()
+
+    def filter(self, record) -> bool:
+        if dist.is_available() and dist.is_initialized():
+            return dist.get_rank() in self.ranks
+
+        return True
+
+
+def _default_filter() -> Optional[TorchLogsFilter]:
+    ranks_str = os.environ.get(LOG_RANK_ENV_VAR, None)
+    if ranks_str is None:
+        return None
+
+    ranks = [int(rank_str) for rank_str in ranks_str.split(",")]
```

Comment:
maybe `try:` around  this parsing bit, and then in the 'except' clause you can raise the ValueError you have?